pkg/scalers/scalersconfig/typed_config.go

```diff
@@ -273,6 +273,22 @@ func setConfigValueURLParams(params Params, valFromConfig string, field reflect.
 	return nil
 }
 
+// setConfigValueURL is a function that sets the value of the url.URL field
+func setConfigValueURL(_ Params, valFromConfig string, field reflect.Value) error {
+	u, err := url.Parse(valFromConfig)
+	if err != nil {
+		return fmt.Errorf("expected url.URL or *url.URL, unable to parse %q: %w", valFromConfig, err)
+	}
+
+	// If the field type is a pointer to url.URL (`*url.URL`), set the value directly
+	if field.Kind() == reflect.Ptr && field.Type().Elem() == reflect.TypeOf(url.URL{}) {
+		field.Set(reflect.ValueOf(u))
+		return nil
+	}
+
+	return nil
+}
```

Comment:
can you please add a simple unit test for this? you can take an inspiration in
https://github.com/kedacore/keda/blob/5b7a372bf249ad6a2a280ee71efc1677b098b04e/pkg/scalers/scalersconfig/typed_config_test.go#L27

Reply:
Done